src/Microsoft.AspNet.Http/QueryCollection.cs

```diff
@@ -0,0 +1,412 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNet.Http.Internal
+{
+    /// <summary>
+    /// The HttpRequest query string collection
+    /// </summary>
+    public sealed class QueryCollection : IQueryCollection
```

Comment:
Have changed the interfaces to be readonly, want me to remove the class direct functions also?